runtime/shared_common/ClasspathItem.cpp

```diff
@@ -56,6 +56,15 @@ ClasspathEntryItem::initialize(const char* path_, U_16 pathLen_, UDATA protocol_
 
 	path = (char*)path_;
 	pathLen = pathLen_;
+	locationPathLen = pathLen_;
+	if (protocol == PROTO_JAR) {
+		if (NULL != path) {
+			char* jarPath = strstr(path,".jar!/");
```

Comment:
Should also support .zip

Reply:
On second look, can this be made more robust by just checking for `!/`

Reply:
Changed to check "!/"